### ISSUE TITLE
JPA Entity check makes incorrect assumption about getter method names.

### ISSUE THREAD

**Issue by sithmein** [jqno/equalsverifier#829]
I like the new JPA entity check for using getters in the equal method instead of the instance variables directly. However, it makes assumptions about the naming of members methods that are not always correct.
We have a coding convention where instance variables are prefixed with `m_`, e.g. `m_groups`. The getter method is then derived as `getM_groups` which is wrong (it's `getGroups` as one would expect), see https://github.com/jqno/equalsverifier/blob/2a8295272462e2e9ede533517429f3314f7c5c02/equalsverifier-core/src/main/java/nl/jqno/equalsverifier/internal/checkers/fieldchecks/JpaLazyGetterFieldCheck.java#L44.
In the end this leads to the incorrect error message `-> JPA Entity: direct reference to field m_groups used in equals instead of getter getM_groups`. Which, when looking at the `equals`-implementation is just wrong.

In order to fix this issue I suggest adding a configuration option where you can specify an optional prefix for instance variables which is then removed from the name when deriving the getter method, e.g.

```
EqualsVerifier.forClass(Foo.class).withMemberVariablePrefix("m_")...
```

**Comment by jqno** [comment]
I see your point. I'm not a big fan of these kinds of conventions, but sometimes we don't get the choice...

I'll have to think a bit about the best way to support this. Maybe it isn't always a prefix, would it be better to provide a function instead of a prefix?

I'll let you know when I come up with something. If you have more thoughts, they're certainly welcome.

**Comment by sithmein** [comment]
You mean a function that the user provides which derives the method name from the member variable name? That would also work and even be more flexible.

**Comment by jqno** [comment]
I've just released version 3.15. It has a `withFieldnameToGetterConverter` method.

This should solve your situation:

```java
EqualsVerifier
    .forClass(Foo.class)
    .withFieldnameToGetterConverter(
        fn -> "get" + Character.toUpperCase(fn.charAt(2)) + fn.substring(3)
    )
    .verify();
```

**Comment by sithmein** [comment]
Thanks a lot! I will give it a try soon.
